### PR TITLE
fix: correct slot and index in getEpochCommittees response

### LINKED ISSUE #5632: /eth/v1/beacon/states/head/committees returns incorrect information
### Describe the bug

Calling `/eth/v1/beacon/states/head/committees` on the API endpoint always returns committees for slot 0, for example on the current deneb testnet:

```
curl -s 'https://xxx@bn.lodestar-ethereumjs-1.srv.4844-devnet-6.ethpandaops.io/eth/v1/beacon/states/head/committees'| jq -r .data[].slot | sort | uniq -c
     32 0
```

### Expected behavior

Would expect to see the committees for the current epoch, for example:

```
curl -s 'https://xxx@bn.teku-nethermind-1.srv.4844-devnet-6.ethpandaops.io/eth/v1/beacon/states/head/committees'| jq -r .data[].slot | sort | uniq -c
      1 38176
      1 38177
      1 38178
      1 38179
      1 38180
      1 38181
      1 38182
      1 38183
      1 38184
      1 38185
      1 38186
      1 38187
      1 38188
      1 38189
      1 38190
      1 38191
      1 38192
      1 38193
      1 38194
      1 38195
      1 38196
      1 38197
      1 38198
      1 38199
      1 38200
      1 38201
      1 38202
      1 38203
      1 38204
      1 38205
      1 38206
      1 38207
```

### Steps to reproduce

As per above, connected to pre-devnet 6

### Additional context

_No response_

### Operating system

Linux

### Lodestar version or commit hash

Whatever is running on the servers as mentioned above

### PR DESCRIPTION
**Pull request by nflaig**
**Motivation**

Closes https://github.com/ChainSafe/lodestar/issues/5632

**Description**

- correct slot and committee index in [getEpochCommittees](https://ethereum.github.io/beacon-APIs/#/Beacon/getEpochCommittees) response
- fix filtering by slot and index

### CODE REVIEW

**Review by naiba4 — APPROVED** [review]

File: packages/beacon-node/test/e2e/api/impl/beacon/state/endpoint.test.ts
Comment (nflaig) [inline packages/beacon-node/test/e2e/api/impl/beacon/state/endpoint.test.ts:1]: tests are aligned with light client api tests https://github.com/ChainSafe/lodestar/blob/82914a1443fa1077d2a217cda9a1248e03084c46/packages/beacon-node/test/e2e/api/impl/lightclient/endpoint.test.ts#L17

File: packages/beacon-node/test/e2e/api/impl/beacon/state/endpoint.test.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import {createBeaconConfig} from "@lodestar/config";
+import {chainConfig as chainConfigDef} from "@lodestar/config/default";
+import {Api, ApiError, getClient} from "@lodestar/api";
+import {computeCommitteeCount} from "@lodestar/state-transition";
+import {LogLevel, testLogger} from "../../../../../utils/logger.js";
+import {getDevBeaconNode} from "../../../../../utils/node/beacon.js";
+import {BeaconNode} from "../../../../../../src/node/nodejs.js";
+
+describe("beacon state api", function () {
+  this.timeout("10 min");
```
Comment (nazarhussain) [inline packages/beacon-node/test/e2e/api/impl/beacon/state/endpoint.test.ts]: This is too aggressive, can we make it more reasonable?

**Review by nazarhussain — DISMISSED** [review]

File: packages/beacon-node/test/e2e/api/impl/beacon/state/endpoint.test.ts (hunk 1 above)
Comment (nflaig) [reply packages/beacon-node/test/e2e/api/impl/beacon/state/endpoint.test.ts]: yeah, might be good idea to reduce that, just copied the value from https://github.com/ChainSafe/lodestar/blob/82914a1443fa1077d2a217cda9a1248e03084c46/packages/beacon-node/test/e2e/api/impl/lightclient/endpoint.test.ts#L18

Comment (nflaig) [inline packages/beacon-node/test/e2e/api/impl/beacon/state/endpoint.test.ts]:
```suggestion
  this.timeout("30s");
```

**Review by nazarhussain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.9.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.9.0) :tada:

**Pull request merged**